common/lib/dependabot/metadata_finders/base/release_finder.rb

```diff
@@ -252,36 +252,38 @@ def previous_version
           end
 
           if dependency.previous_version.match?(/^[0-9a-f]{40}$/) &&
-             ref_changed?
+             ref_changed? && previous_ref
             previous_ref
           else
             dependency.previous_version
           end
         end
 
         def new_version
-          if dependency.version.match?(/^[0-9a-f]{40}$/) && ref_changed?
+          if dependency.version.match?(/^[0-9a-f]{40}$/) &&
+              ref_changed? && new_ref
```

Comment:
Not sure this actually possible when the version is a git sha but seems like we could go from a `previous_ref` to `nil` `new_ref` here